app/controllers/posts_controller.rb

```diff
@@ -20,7 +20,11 @@ def create
   private
 
   def timeline_posts
-    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
+    us = current_user.friends.map(&:id)
+    us << current_user.id
+    us = us.join(',')
+    first_part = 'SELECT * FROM posts WHERE user_id IN'
+    @timeline_posts = Post.find_by_sql("#{first_part} (SELECT id FROM users WHERE id IN(#{us})) ORDER BY created_at")
```

Comment:
- [ ] Remember to keep model logic, ActiveRecord and SQL queries, inside model methods and not directly in the controllers.